query/stdlib/influxdata/influxdb/rules.go

```diff
@@ -764,3 +767,101 @@ func (PushDownWindowAggregateRule) Rewrite(ctx context.Context, pn plan.Node) (p
 		WindowEvery:       window.Every.Nanoseconds(),
 	}), true, nil
 }
+
+//
+// Push Down of group  aggregates.
+// ReadGroupPhys |> { min, max, mean, count, sum }
+//
+type PushDownGroupAggregateRule struct{}
+
+func (PushDownGroupAggregateRule) Name() string {
+	return "PushDownWindowAggregateRule"
+}
+
+func (rule PushDownGroupAggregateRule) Pattern() plan.Pattern {
+	return plan.OneOf(
+		[]plan.ProcedureKind{
+			universe.MinKind,
+			universe.MaxKind,
+			universe.MeanKind,
+			universe.CountKind,
+			universe.SumKind,
+		},
+		plan.Pat(ReadGroupPhysKind))
+}
+
+func (PushDownGroupAggregateRule) Rewrite(ctx context.Context, pn plan.Node) (plan.Node, bool, error) {
+	// Check the aggregate function spec. Require operation on _value. There
+	// are two feature flags covering count and sum. All the other cases
+	// are specifically turned off.
```

Comment:
Discussed this with Nathaniel and he didn't want rule cases present without feature flags covering them.

I understand the desire to not populate the code with cases we don't have implementations for so how about we leave these here since they are tested and in future rules we refrain from added extra cases?

Reply:
It looks like we will be asked to implement these cases soon that's another reason to leave them in for now. At the end of this push we can delete the stuff that won't be done in the immediate future.

Reply:
I'll add an issue to expand on OneOf testing so we can remove cases that aren't used, without also reducing test coverage.